aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -0,0 +1,295 @@
+
+    private final ModelSchema modelSchema;
+    private final OperationType operationType;
+    private final SelectionSet.Node selectionSet;
+    private final Map<String, Object> variables;
+    private final Map<String, String> variableTypes;
+
+    /**
+     * Constructor for AppSyncGraphQLRequest.
+     * @throw AmplifyException if a ModelSchema can't be derived from the provided model class
+     */
+    private AppSyncGraphQLRequest(Builder builder) throws AmplifyException {
+        super(builder.responseType, new GsonVariablesSerializer());
+        this.modelSchema = ModelSchema.fromModelClass(builder.modelClass);
+        SelectionSet.Node set = SelectionSet.fromModelClass(builder.modelClass, DEFAULT_LEVEL_DEPTH);
+        if (QueryType.LIST.equals(builder.operationType)) {
+            set = SelectionSet.wrapPagination(set);
+        }
+        this.selectionSet = set;
+        this.operationType = builder.operationType;
+        this.variables = builder.variables;
+        this.variableTypes = builder.variableTypes;
+    }
+
+    /**
+     * Copy constructor for an AppSyncGraphQLRequest.
+     * @param request request to copy.
+     * @param <R> response type.
+     */
+    public <R> AppSyncGraphQLRequest(AppSyncGraphQLRequest<R> request) {
+        super(request);
+        this.modelSchema = request.modelSchema;
+        this.operationType = request.operationType;
+        this.selectionSet = new SelectionSet.Node(request.selectionSet);
+        this.variables = new HashMap<>(request.variables);
+        this.variableTypes = new HashMap<>(request.variableTypes);
+    }
+
+    @Override
+    public <R> AppSyncGraphQLRequest<R> copy() {
+        return new AppSyncGraphQLRequest<R>(this);
+    }
+
+    @Override
+    public Map<String, Object> getVariables() {
+        return this.variables;
+    }
+
+    /**
+     * Used for setting a variable on the request.
+     * @param key variable name
+     * @param type type of variable value
+     * @param value variable value
+     */
+    public void setVariable(String key, String type, String value) {
+        this.variables.put(key, value);
+        this.variableTypes.put(key, type);
+    }
+
+    /**
+     * Used to add owner argument from authProvider if needed.
+     * @param authProvider CognitoUserPoolsAuthProvider for obtaining the username to set as the owner field.
+     * @throws ApiException if request requires owner argument and authProvider or authProvider.getUsername() is null.
+     */
+    public void setAuthProvider(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
+        for (AuthRule authRule : modelSchema.getAuthRules()) {
+            if (isOwnerArgumentRequired(authRule.getOperationsOrDefault())) {
+                setVariable(authRule.getOwnerFieldOrDefault(), "String!", getUsername(authProvider));
+            }
+        }
+    }
+
+    private String getUsername(CognitoUserPoolsAuthProvider authProvider) throws ApiException {
```

Comment:
This is what iOS is currently doing as well, using the username as the identityClaim and we have opened a [feature request](https://github.com/aws-amplify/amplify-ios/issues/485) to support more than just the default identityClaim type when using the auth directive